Improve error message for "helios deploy <jobID> ."

`helios deploy -z tcp://<helios-master>:5801 <jobID> . -vv` returns the error below when using "." for the helios agent.

```
14:03:20.974 DEBUG HeliosClient: endpoint uris are [tcp://<helios-master>:5801]
14:03:20.978 DEBUG HeliosClient: connecting to http://<helios-master>:5801/jobs?q=<jobID>&user=davidxia
14:03:20.978 DEBUG HeliosClient: req: GET http://<helios-master>:5801/jobs?q=<jobID>&user=davidxia 1 0
14:03:22.107 DEBUG HeliosClient: rep: GET http://<helios-master>:5801/jobs?q=<jobID>&user=davidxia 200 726 gzip:true
Deploying Deployment{jobId=<jobID>07369ffa45281042c8646cd71217be40b, goal=START, deployerUser=null} on [.]
14:03:22.227 DEBUG HeliosClient: endpoint uris are [tcp://<helios-master>:5801]
14:03:22.228 DEBUG HeliosClient: connecting to http://<helios-master>:5801/hosts/?user=davidxia
14:03:22.228 DEBUG HeliosClient: req: GET http://<helios-master>:5801/hosts/?user=davidxia 1 0
14:03:22.242 DEBUG HeliosClient: rep: GET http://<helios-master>:5801/hosts/?user=davidxia 200 6619 gzip:true
.: 14:03:22.283 DEBUG HeliosClient: endpoint uris are [tcp://<helios-master>:5801]
14:03:22.283 DEBUG HeliosClient: connecting to http://<helios-master>:5801/hosts/./jobs/<jobID>07369ffa45281042c8646cd71217be40b?token=&user=davidxia
14:03:22.284 DEBUG HeliosClient: req: PUT http://<helios-master>:5801/hosts/./jobs/<jobID>07369ffa45281042c8646cd71217be40b?token=&user=davidxia 3 123
14:03:22.303 DEBUG HeliosClient: rep: PUT http://<helios-master>:5801/hosts/./jobs/<jobID>07369ffa45281042c8646cd71217be40b?token=&user=davidxia 500 99 gzip:false
java.lang.RuntimeException: com.spotify.helios.common.HeliosException: request failed: Response{method='PUT', uri=http://helios/hosts/./jobs/<jobID>07369ffa45281042c8646cd71217be40b?token=&user=davidxia, status=500, payload={"message":"There was an error processing your request. It has been logged (ID d681f4ee8b080cd6)."}}
  at com.google.common.base.Throwables.propagate(Throwables.java:160)
  at com.spotify.helios.cli.command.ControlCommand.run(ControlCommand.java:132)
  at com.spotify.helios.cli.command.ControlCommand.run(ControlCommand.java:89)
  at com.spotify.helios.cli.CliMain.run(CliMain.java:79)
  at com.spotify.helios.cli.CliMain.main(CliMain.java:57)
Caused by: com.spotify.helios.common.HeliosException: request failed: Response{method='PUT', uri=http://helios/hosts/./jobs/<jobID>07369ffa45281042c8646cd71217be40b?token=&user=davidxia, status=500, payload={"message":"There was an error processing your request. It has been logged (ID d681f4ee8b080cd6)."}}
  at com.spotify.helios.client.HeliosClient$ConvertResponseToPojo.apply(HeliosClient.java:590)
  at com.spotify.helios.client.HeliosClient$ConvertResponseToPojo.apply(HeliosClient.java:558)
  at com.google.common.util.concurrent.Futures$ChainingListenableFuture.run(Futures.java:863)
  at com.google.common.util.concurrent.MoreExecutors$SameThreadExecutorService.execute(MoreExecutors.java:297)
  at com.google.common.util.concurrent.ExecutionList.executeListener(ExecutionList.java:156)
  at com.google.common.util.concurrent.ExecutionList.execute(ExecutionList.java:145)
  at com.google.common.util.concurrent.ListenableFutureTask.done(ListenableFutureTask.java:91)
  at java.util.concurrent.FutureTask$Sync.innerSet(FutureTask.java:281)
  at java.util.concurrent.FutureTask.set(FutureTask.java:141)
  at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:339)
  at java.util.concurrent.FutureTask.run(FutureTask.java:166)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
  at java.lang.Thread.run(Thread.java:724)
```

The helios master's error log shows

```
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]: Error handling a request: 56a6f9260d7e47ed
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   java.lang.IllegalArgumentException: Invalid path string "/config/hosts/./jobs/<jobID>07369ffa45281042c8646cd71217be40b" caused by relative paths not allowed @14
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.apache.zookeeper.common.PathUtils.validatePath(PathUtils.java:99)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.spotify.helios.servicescommon.coordination.PathFactory.path(PathFactory.java:38)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.spotify.helios.servicescommon.coordination.Paths.configHostJob(Paths.java:113)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.spotify.helios.master.ZooKeeperMasterModel.deployJobRetry(ZooKeeperMasterModel.java:569)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.spotify.helios.master.ZooKeeperMasterModel.deployJob(ZooKeeperMasterModel.java:545)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.spotify.helios.master.resources.HostsResource.jobPut(HostsResource.java:202)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at sun.reflect.GeneratedMethodAccessor32.invoke(Unknown Source)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at java.lang.reflect.Method.invoke(Method.java:606)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.spi.container.JavaMethodInvokerFactory$1.invoke(JavaMethodInvokerFactory.java:60)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.model.method.dispatch.AbstractResourceMethodDispatchProvider$TypeOutInvoker._dispatch(AbstractResourceMethodDispatchProvider.java:185)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.model.method.dispatch.ResourceJavaMethodDispatcher.dispatch(ResourceJavaMethodDispatcher.java:75)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.codahale.metrics.jersey.InstrumentedResourceMethodDispatchProvider$TimedRequestDispatcher.dispatch(InstrumentedResourceMethodDispatchProvider.java:30)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.codahale.metrics.jersey.InstrumentedResourceMethodDispatchProvider$ExceptionMeteredRequestDispatcher.dispatch(InstrumentedResourceMethodDispatchProvider.java:69)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.spotify.helios.master.metrics.ReportingResourceMethodDispatcher.dispatch(ReportingResourceMethodDispatcher.java:52)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at io.dropwizard.jersey.guava.OptionalResourceMethodDispatchAdapter$OptionalRequestDispatcher.dispatch(OptionalResourceMethodDispatchAdapter.java:37)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.uri.rules.HttpMethodRule.accept(HttpMethodRule.java:302)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.uri.rules.ResourceObjectRule.accept(ResourceObjectRule.java:100)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.uri.rules.RootResourceClassesRule.accept(RootResourceClassesRule.java:84)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1542)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1473)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1419)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1409)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:409)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:540)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:715)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at io.dropwizard.jetty.NonblockingServletHolder.handle(NonblockingServletHolder.java:49)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1515)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlets.UserAgentFilter.doFilter(UserAgentFilter.java:83)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlets.GzipFilter.doFilter(GzipFilter.java:295)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at io.dropwizard.jetty.BiDiGzipFilter.doFilter(BiDiGzipFilter.java:127)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at io.dropwizard.servlets.ThreadNameFilter.doFilter(ThreadNameFilter.java:29)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at io.dropwizard.jersey.filter.AllowedMethodsFilter.handle(AllowedMethodsFilter.java:44)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at io.dropwizard.jersey.filter.AllowedMethodsFilter.doFilter(AllowedMethodsFilter.java:39)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlets.CrossOriginFilter.handle(CrossOriginFilter.java:248)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlets.CrossOriginFilter.doFilter(CrossOriginFilter.java:211)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.spotify.helios.master.http.VersionResponseFilter.doFilter(VersionResponseFilter.java:99)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1486)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:519)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1097)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:448)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1031)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:136)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at com.codahale.metrics.jetty9.InstrumentedHandler.handle(InstrumentedHandler.java:175)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at io.dropwizard.jetty.RoutingHandler.handle(RoutingHandler.java:51)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:162)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:109)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.Server.handle(Server.java:446)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:271)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:246)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.io.AbstractConnection$ReadCallback.run(AbstractConnection.java:358)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:601)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:532)
2015-04-20T18:04:12.518+00:00 <helios-master> helios[32211]:   at java.lang.Thread.run(Thread.java:744)
```

`IllegalArgumentException` is thrown at `final String taskPath = Paths.configHostJob(host, id)` in `ZooKeeperMasterModel`. We should throw `HostNotFoundException` somewhere there for `HostsResource:jobPut()` to catch later on.

--- a/helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java
+++ b/helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java
@@ -566,6 +566,13 @@ public class ZooKeeperMasterModel implements MasterModel {
 
     final UUID operationId = UUID.randomUUID();
     final String jobPath = Paths.configJob(id);
+
+    try {
+      Paths.configHostJob(host, id);
+    } catch (IllegalArgumentException e) {
+      throw new HostNotFoundException("Could not find Helios host '" + host + "'");
+    }
+
     final String taskPath = Paths.configHostJob(host, id);
     final String taskCreationPath = Paths.configHostJobCreation(host, id, operationId);
 

